common/persistence/visibility/store/standard/cassandra/visibility_store.go

```diff
@@ -433,13 +433,22 @@ func (v *visibilityStore) ListClosedWorkflowExecutionsByStatus(
 }
 
 func (v *visibilityStore) DeleteWorkflowExecution(request *manager.VisibilityDeleteWorkflowExecutionRequest) error {
-	query := v.session.Query(templateDeleteWorkflowExecutionClosed,
-		request.NamespaceID.String(),
-		namespacePartition,
-		persistence.UnixMilliseconds(request.CloseTime),
-		request.RunID).
-		Consistency(v.lowConslevel)
-	if err := query.Exec(); err != nil {
+	var query gocql.Query
+	if request.StartTime != nil {
+		query = v.session.Query(templateDeleteWorkflowExecutionStarted,
+			request.NamespaceID.String(),
+			namespacePartition,
+			persistence.UnixMilliseconds(*request.StartTime),
+			request.RunID)
+	} else if request.CloseTime != nil {
+		query = v.session.Query(templateDeleteWorkflowExecutionClosed,
+			request.NamespaceID.String(),
+			namespacePartition,
+			persistence.UnixMilliseconds(*request.CloseTime),
+			request.RunID)
+	}
+
+	if err := query.Consistency(v.lowConslevel).Exec(); err != nil {
```

Comment:
This will throw NPE if both `StartTime` and `CloseTime` are `nil`. I know it is not possible but let's panic with more explicit message like
```
panic("both StartTime and CloseTime are nil")
```